lib/filesystem.js

```diff
@@ -116,13 +131,25 @@ exports.writeChunk = (fd, chunk, position, callback, retries = 1) => {
  */
 exports.readChunk = (fd, size, position, callback, retries = 1) => {
   fs.read(fd, Buffer.allocUnsafe(size), 0, size, position, (error, bytesRead, buffer) => {
-
     if (error) {
+      const platform = os.platform();
```

Comment:
Given the similarity, would there be any scope for `.writeChunk` and `.readChunk` to share parts of their error-handling?
Or does it makes sense for them to remain distinct, given that read errors and write errors in theory could be quite different?